Adyen/Core/Payment Methods/Abstract/AnyPaymentMethodDecoder.swift

```diff
@@ -491,17 +491,26 @@ private struct TwintPaymentMethodDecoder: PaymentMethodDecoder {
 
     func decode(from decoder: Decoder, isStored: Bool) throws -> AnyPaymentMethod {
         #if canImport(TwintSDK)
-            try .twint(TwintPaymentMethod(from: decoder))
+            if isStored {
+                try .storedTwint(StoredTwintPaymentMethod(from: decoder))
+            } else {
+                try .twint(TwintPaymentMethod(from: decoder))
+            }
         #else
             return AnyPaymentMethod(InstantPaymentMethod(type: .twint, name: "Twint"))
         #endif
     }
 
     func anyPaymentMethod(from paymentMethod: any PaymentMethod) -> AnyPaymentMethod? {
         #if canImport(TwintSDK)
-            (paymentMethod as? TwintPaymentMethod).map { .twint($0) }
-        #else
-            return nil
+            if let method = paymentMethod as? TwintPaymentMethod {
```

Comment:
Good question!
If that happens, we're on a point where we know it's a valid payment method since we were able to decode it.
With that being said, it means it is a payment method that we don't recognize in the SDK so we treat it like an instant payment method.

You can trace this logic all the way to `AnyPaymentMethodDecoder.swift:134`